ktor-client/ktor-client-core/common/src/io/ktor/client/features/DefaultResponseValidation.kt

```diff
@@ -47,7 +47,8 @@ fun HttpClientConfig<*>.addDefaultResponseValidation() {
 open class ResponseException(
     response: HttpResponse
 ) : IllegalStateException("Bad response: $response") {
-    public val response: HttpResponse? by threadLocal(response)
+    private val _response: HttpResponse? by threadLocal(response)
+    public val response: HttpResponse get() = _response ?: error("Failed to access response from a different native thread")
```

Comment:
How do you know that `null` in this case means that was a failed attempt of access from a different native thread?

Reply:
It has thread-local behavior only in the case of kotlin-native

Reply:
Then it would be great to document this assumption